### ISSUE TITLE
Invalid Move Player Packet Recieved

### ISSUE THREAD

**Issue by FireController1847** [bergerhealer/TrainCarts#307]
##### Spigot version: 1.14.4-R0.1-SNAPSHOT `git-Spigot-94af569-a3fdef8 `

<hr>

#### Problem or bug:
TrainCarts corrupts playerdata when destroying a furnace minecart while inside a regular minecart

#### Expected behaviour:
It should not do that

#### Steps to reproduce:
Set maxVelocity to 1.0 blocks / tick
Get a coal-powered minecart and attach a regular minecart to it
Get it up to speed in a moderately sized loop
Destroy the coal-powered minecart while still inside the regular minecart
Watch as your NBT data is corrupted and you can no longer connect to the server
To rejoin, you have to delete your playerdata in the world folder

**Comment by bergerkiller** [comment]
is destroying the furnace minecart really required to trigger this bug? Do you leave the server while still inside the minecart? What is 'allowPlayerTake ' set to in DefaultTrainProperties.yml?

**Comment by weeryan17** [comment]
I'm unable to reproduce this with just traincarts and bkcommonlib installed on my server.

**Comment by bergerkiller** [comment]
https://github.com/bergerhealer/TrainCarts/blob/master/src/main/java/com/bergerkiller/bukkit/tc/controller/type/MinecartMemberFurnace.java#L40 potential spot for push x/z to be set to either NaN or Infinite

**Comment by FireController1847** [comment]
Sometimes, when the player destroys the cart, the cart the player is in will teleport into the ground. I've noticed the game freezes for a second, you'll see into the world and glitch back and forth (you can't exit the minecart), and then it kicks you out with the error. Any further attempts to rejoin result in "Internal Server Error" with the following crash:
```

[11:48:26] [Server thread/WARN]: Failed to handle packet for /xx.xxx.xx.xxx:xxxxx
net.minecraft.server.v1_14_R1.ReportedException: Loading entity NBT
        at net.minecraft.server.v1_14_R1.Entity.f(Entity.java:1727) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.WorldNBTStorage.load(WorldNBTStorage.java:204) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.PlayerList.a(PlayerList.java:352) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.PlayerList.a(PlayerList.java:102) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.LoginListener.c(LoginListener.java:154) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.LoginListener.tick(LoginListener.java:53) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.NetworkManager.a(NetworkManager.java:220) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.ServerConnection.c(ServerConnection.java:129) [spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.b(MinecraftServer.java:1091) [spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.DedicatedServer.b(DedicatedServer.java:393) [spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:970) [spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:815) [spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.IllegalStateException: Entity has invalid position
        at net.minecraft.server.v1_14_R1.Entity.f(Entity.java:1681) ~[spigot-1.14.4.jar:git-Spigot-94af569-a3fdef8]
        ... 12 more
```

**Comment by bergerkiller** [comment]
See if you can still reproduce it with this fix:
https://github.com/bergerhealer/TrainCarts/commit/07d0f55727083228e968f7dd50208c1b50ec2791

https://ci.mg-dev.eu/job/TrainCarts/519/

**Comment by FireController1847** [comment]
I have yet to have this issue again even on the latest version, so I'm now questioning even my own reproducability steps. I have yet to have it happen, so I'll say it's fixed, but I'll continue testing and using 2.519 and re-open this again! :)
